File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/director_detect_runme.pl
use strict;
use warnings;
use Test::More tests => 9;
BEGIN { use_ok 'director_detect' }
require_ok 'director_detect';

{
  package MyBar;
  use base 'director_detect::Bar';
  sub new { my $class = shift;
    my $val = @_ ? shift : 2;
    my $self = $class->SUPER::new();
    $self->{val} = $val;
    return $self;
  }
  sub get_value { my($self) = @_;
    $self->{val}++;
    return $self->{val};
  }
  sub get_class { my($self) = @_;
    $self->{val}++;
    return director_detect::A->new();
  }
  sub just_do_it { my($self) = @_;
    $self->{val}++;
  }
  sub clone { my($self) = @_;
    MyBar->new($self->{val});
  }
}

my $b = MyBar->new();
isa_ok $b, 'MyBar';

my $f = $b->baseclass();
isa_ok $f, 'director_detect::Foo';
is $f->get_value(), 3;

isa_ok $f->get_class(), 'director_detect::A';
$f->just_do_it();

my $c = $b->clone();
isa_ok $c, 'MyBar';
is $b->{val}, 5;
is $c->get_value(), 6;

File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/director_detect_runme.m
director_detect 

global MyBar=@(val=2) subclass(director_detect.Bar(),'val',val,@get_value,@get_class,@just_do_it,@clone);

function val=get_value(self)
    self.val = self.val + 1;
    val = self.val;
end
function ptr=get_class(self)
  global director_detect;
  self.val = self.val + 1;
  ptr=director_detect.A();
end
function just_do_it(self)
  self.val = self.val + 1;
end
function ptr=clone(self)
  global MyBar;
  ptr=MyBar(self.val);
end

b = MyBar();

f = b.baseclass();

v = f.get_value();
a = f.get_class();
f.just_do_it();

c = b.clone();
vc = c.get_value();

if ((v != 3) || (b.val != 5) || (vc != 6))
  error("Bad virtual detection")
endif

